app/forms/Movement.scala

```diff
@@ -16,27 +16,56 @@
 
 package forms
 
-import forms.Choice._
-import models.cache.Cache
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+
+import models.ReturnToStartException
+import models.cache.{Answers, ArrivalAnswers, DepartureAnswers}
 import models.requests.{MovementDetailsRequest, MovementRequest, MovementType}
 
 object Movement {
 
-  def createMovementRequest(cache: Cache): MovementRequest =
-    // TODO - implement
+  private val departureDateTimeFormatter = DateTimeFormatter.ISO_INSTANT
+
+  def createMovementRequest(pid: String, answers: Answers): MovementRequest = answers match {
+    case arrivalAnswers: ArrivalAnswers     => createMovementArrivalRequest(pid, arrivalAnswers)
+    case departureAnswers: DepartureAnswers => createMovementDepartureRequest(pid, departureAnswers)
+  }
+
+  private def createMovementArrivalRequest(pid: String, answers: ArrivalAnswers) =
     MovementRequest(
-      eori = "TODO",
+      eori = answers.eori.getOrElse(throw ReturnToStartException),
+      providerId = Some(pid),
       choice = MovementType.Arrival,
-      consignmentReference = ConsignmentReferences("todo", "todo"),
-      movementDetails = MovementDetailsRequest("todo"),
-      location = None,
-      transport = None,
-      arrivalReference = None
+      consignmentReference = answers.consignmentReferences.getOrElse(throw ReturnToStartException),
+      movementDetails = movementDetails(answers),
+      location = answers.location,
+      arrivalReference = answers.arrivalReference
     )
 
-  private def extractChoice(choice: Choice) = choice match {
-    case Arrival   => MovementType.Arrival
-    case Departure => MovementType.Departure
-    case _         => throw new IllegalArgumentException("Allowed is only arrival or departure here")
-  }
+  private def createMovementDepartureRequest(pid: String, answers: DepartureAnswers) =
+    MovementRequest(
+      eori = answers.eori.getOrElse(throw ReturnToStartException),
+      providerId = Some(pid),
```

Comment:
We can just change the MovementRequest.pid to be a mandatory field not optional, it will always be present

Reply:
Again - just porting over code.

Reply:
Made non-optional